pkg/prometheus/promcfg.go

```diff
@@ -781,6 +815,8 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 	}
 	if ep.TLSConfig != nil {
 		cfg = addSafeTLStoYaml(cfg, m.Namespace, ep.TLSConfig.SafeTLSConfig)
+	} else if scrapeClass != nil {
```

Comment:
the proposal stated that the TLS configs would be merged with the monitor object's fields taking precedence.